_posts/2024-02-27-comet-donation.md

```diff
@@ -0,0 +1,106 @@
+accelerate Spark workloads. It is designed as a drop-in
+replacement for Spark's JVM based SQL execution engine and offers significant
+performance improvements for some workloads as shown below.
+
+```text
+   ┌─────────────────────────────────────────────────────────────────┐
+   │                                                                 │
+   │ ┌──────────┐ ┌────────────┐ ┌────────────┐       ┌────────────┐ │
+   │ │   SQL    │ │  Cluster   │ │  DAG/Task  │  ...  │  Executor  │ │
+   │ │ Planner  │ │  Manager   │ │ Scheduler  │       │            │ │
+   │ └──────────┘ └────────────┘ └────────────┘       └────────────┘ │
+   │                                                         │       │
+   └─────────────────────────────────────────────────────────────────┘
+     Spark (JVM Based)                                       │        
+                                  ┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─         
+                                                                      
+                                  │                                   
+                                  ▼                                   
+                 ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓                   
+Comet Execution  ┃                                ┃                   
+Engine           ┃  ┌─────────────────────────┐   ┃                   
+(Native Code)    ┃  │ Apache Arrow DataFusion │   ┃                   
+                 ┃  └─────────────────────────┘   ┃                   
+                 ┃                                ┃                   
+                 ┃  ┌─────────────────────────┐   ┃                   
+                 ┃  │    Spark Compatible     │   ┃                   
+                 ┃  │  Expressions/Operators  │   ┃                   
+                 ┃  └─────────────────────────┘   ┃                   
+                 ┃                                ┃                   
+                 ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛                   
+```
+
+**Figure 1**: With Comet, users interact with the same Spark ecosystem, tools
+and APIs such as Spark SQL. Queries still run through Spark's mature and feature
+rich query optimizer and planner. However, the execution is delegated to Comet,
+which is significantly faster and more resource efficient than the  JVM based
+implementation.
+
+[Rust]: https://www.rust-lang.org/
+
+# Background
+
+Comet is one of a growing class of projects that aim to accelerate Spark using
+native columnar engines such as the proprietary [Databricks Photon Engine] and
+the open source [Gluten project] and [Spark RAPIDS].
+
+Comet was originally implemented at Apple and the engineers who worked on the
+project are also significant contributors to Arrow and DataFusion. Bringing 
+Comet into the Apache Software Foundation will accelerate its development and 
+grow its community of contributors and users.
+
+[Comet project]: https://github.com/apache/arrow-datafusion-comet
```

Comment:
I learned that there is a similar effort to build a Spark-native accelerator using DataFusion: [Blaze](https://github.com/kwai/blaze). Wondering if it deserves a mention as well.

Reply:
@richox or  @yjshen, who worked on Blaze in the past, do you have any suggestions or thoughts on if/how we should mention Blaze in this post?

The last time I checked the Blaze project hadn't had any updates, but it appears that in the last 6 months @richox has been working on it (BTW @richox contributed some excellent  improvements to DataFusion as well such a a loser tree for the sorting implementation 🙏 )

Reply:
Added a link in 919c934

Reply:
I heard the current Blaze team has proposed donating it to Apache in the past, but I am not sure what the status is.

Reply:
It is probably unrealistic, but perhaps the projects could converge over time

Reply:
+1. It would be great if Blaze and Comet can join forces in some way!